galasa-parent/dev.galasa.framework.api.authentication/src/main/java/dev/galasa/framework/api/authentication/internal/OidcProvider.java

```diff
@@ -66,39 +71,44 @@ public class OidcProvider implements IOidcProvider {
     private Instant nextJwkRefresh = Instant.EPOCH;
     private ITimeService timeService;
 
-    private String issuerUrl;
+    private URI issuerUrl;
     private String authorizationEndpoint;
     private String tokenEndpoint;
     private String jwksUri;
 
     private HttpClient httpClient = HttpClient.newHttpClient();
 
-    public OidcProvider(String issuerUrl, HttpClient httpClient, ITimeService timeService) {
-        this.issuerUrl = issuerUrl;
-        this.httpClient = httpClient;
-        this.timeService = timeService;
+    public OidcProvider(String issuerUrl, HttpClient httpClient, ITimeService timeService) throws ServletException {
+        try {
+            this.issuerUrl = new URI(issuerUrl);
+            this.httpClient = httpClient;
+            this.timeService = timeService;
 
-        this.authorizationEndpoint = issuerUrl + "/auth";
-        this.tokenEndpoint = issuerUrl + "/token";
-        this.jwksUri = issuerUrl + "/keys";
+            this.authorizationEndpoint = issuerUrl + "/auth";
+            this.tokenEndpoint = issuerUrl + "/token";
+            this.jwksUri = issuerUrl + "/keys";
 
-        try {
             JsonObject openIdConfiguration = getOpenIdConfiguration();
             if (openIdConfiguration != null) {
                 // The following endpoints are mandatory in OpenID configurations
-                this.authorizationEndpoint = openIdConfiguration.get("authorization_endpoint").getAsString();
-                this.tokenEndpoint = openIdConfiguration.get("token_endpoint").getAsString();
-                this.jwksUri = openIdConfiguration.get("jwks_uri").getAsString();
+                this.authorizationEndpoint = getValidatedUrl(openIdConfiguration.get("authorization_endpoint").getAsString());
+                this.tokenEndpoint = getValidatedUrl(openIdConfiguration.get("token_endpoint").getAsString());
+                this.jwksUri = getValidatedUrl(openIdConfiguration.get("jwks_uri").getAsString());
             }
+
         } catch (IOException | InterruptedException | JsonSyntaxException e) {
             logger.error("Unable to obtain issuer's OpenID configuration, using defaults");
+        } catch (URISyntaxException e) {
+            ServletError error = new ServletError(GAL5059_INVALID_ISSUER_URI_PROVIDED, issuerUrl);
```

Comment:
Passing this Url up for others might be problematic if it contains malicious stuff... and we know it's not valid.
Suggest we log the contents and error without saying what we got.

Reply:
Addressed in https://github.com/galasa-dev/framework/pull/566/commits/8c56f5c11bf27c9e99ea1f836408149fac1295a3